### ISSUE TITLE
Backlog search keeps trying to download episodes already in download client

### ISSUE THREAD

**Issue by fuxxociety** [pymedusa/Medusa#5991]
**Describe the bug**
I keep getting errors in Medusa.
"WARNING  SNATCHQUEUE-SNATCH-248736 :: [918cfe7] Deluge: Unable to send Torrent"

Investigation into the issue on my end shows the backlog search is attempting to download an episode that is already snatched, but not finished downloading. Either Deluge is not providing a proper error message for "Torrent already exists in client", or Medusa is not parsing that error properly.

**To Reproduce**
1. Allow an episode to snatch, and be sent to Deluge.
2. Wait. Wait. Wait some more. Download never finishes and is never marked as failed.
3. Scheduled Backlog search runs automatically. Backlog search successfully finds an episode and sends it to Deluge.
3. Torrent sent to Deluge is rejected, returning "Deluge: Unable to send Torrent"

**Expected behavior**
Ideally, the backlog search should check the torrent client for downloads already in-progress, or marked as "Snatched". Backlog should not be performed on episodes that have already been sent to the torrent client.
If the torrent has been stalled for quite some time - In my case, almost two months - Possibly automatically mark as failed? (This would probably be a separate feature request)

**Screenshots**

**Medusa (please complete the following information):**
 - OS: [Debian 9.5 x64 ESXi Virtual Machine]
 - Branch: [master]
 - Commit: [918cfe7]

**Logs:**
<details>
![deluge window](https://user-images.githubusercontent.com/3860592/51058162-810ab980-15ad-11e9-8439-8aac2d89b180.png)

```
2019-01-11 07:58:30 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search string: Dinotrux Supercharged S02E03
2019-01-11 07:58:30 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Searching only ranked torrents
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] GET URL: http://torrentapi.org/pubapi_v2.php?sort=seeders&min_leechers=0&min_seeders=1&search_tvdb=337661&app_id=med$
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] No results found Code: 20
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search mode: Episode
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search string: Dinotrux Supercharged S02E04
2019-01-11 07:58:36 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Searching only ranked torrents
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] GET URL: http://torrentapi.org/pubapi_v2.php?sort=seeders&min_leechers=0&min_seeders=1&search_tvdb=337661&app_id=med$
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] No results found Code: 20
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search mode: Episode
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search string: Dinotrux Supercharged S02E05
2019-01-11 07:58:41 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Searching only ranked torrents
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] GET URL: http://torrentapi.org/pubapi_v2.php?sort=seeders&min_leechers=0&min_seeders=1&search_tvdb=337661&app_id=med$
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] No results found Code: 20
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search mode: Episode
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Search string: Dinotrux Supercharged S02E06
2019-01-11 07:58:46 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Searching only ranked torrents
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] GET URL: http://torrentapi.org/pubapi_v2.php?sort=seeders&min_leechers=0&min_seeders=1&search_tvdb=337661&app_id=med$
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] No results found Code: 20
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Found 0 unique items
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [Rarbg] :: [918cfe7] Found qualities: []
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [918cfe7] Combining single and multi-episode results
2019-01-11 07:58:51 DEBUG    SEARCHQUEUE-BACKLOG-337661 :: [918cfe7] No results to sort.
2019-01-11 07:58:51 INFO     SEARCHQUEUE-BACKLOG-337661 :: [918cfe7] Downloading www.SceneTime.com.-.Dinotrux.Supercharged.S02E01.1080p.WEB.x264-STRiFE with 2 seeders and 0 leechers and size 886.2$
2019-01-11 07:58:53 INFO     SEARCHQUEUE-BACKLOG-337661 :: [918cfe7] Downloading Dinotrux.Supercharged.S02E03.720p.WEB.x264-STRiFE[rarbg] with 1 seeders and 0 leechers and size 566.35 MB from BTDB
2019-01-11 07:58:54 INFO     SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Beginning to snatch release: www.SceneTime.com.-.Dinotrux.Supercharged.S02E01.1080p.WEB.x264-STRiFE
2019-01-11 07:58:54 INFO     SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Downloading www.SceneTime.com.-.Dinotrux.Supercharged.S02E01.1080p.WEB.x264-STRiFE with 2 seeders and 0 leechers and size 886.26$
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Calling Deluge Client
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": ["**********"], "method": "auth.login", "id": 1}
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": [], "method": "web.connected", "id": 10}
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Requested a POST connection to http://127.0.0.1:8112/json with params: None Data: {"params": ["magnet:?xt=urn:btih:1FE08$
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": ["magnet:?xt=urn:btih:1FE084F3EE575273779902902F14A2EF4A98E3BB&dn=www.SceneTime.com+-+Dinotrux.Superc$
2019-01-11 07:58:54 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Response to POST request is {"id": 2, "result": null, "error": null}
2019-01-11 07:58:54 WARNING  SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Unable to send Torrent
2019-01-11 07:58:55 INFO     SEARCHQUEUE-BACKLOG-337661 :: [918cfe7] Downloading Dinotrux.Supercharged.S02E05.720p.WEB.x264-STRiFE[ettv] with 1 seeders and 0 leechers and size 592.26 MB from BTDB
2019-01-11 07:58:57 INFO     SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Beginning to snatch release: Dinotrux.Supercharged.S02E03.720p.WEB.x264-STRiFE[rarbg]
2019-01-11 07:58:57 INFO     SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Downloading Dinotrux.Supercharged.S02E03.720p.WEB.x264-STRiFE[rarbg] with 1 seeders and 0 leechers and size 566.35 MB from BTDB,$
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Calling Deluge Client
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": ["**********"], "method": "auth.login", "id": 1}
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": [], "method": "web.connected", "id": 10}
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Requested a POST connection to http://127.0.0.1:8112/json with params: None Data: {"params": ["magnet:?xt=urn:btih:84C92$
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] POST URL: http://127.0.0.1:8112/json [Status: 200]
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] User-Agent: Medusa/0.2.14 (Linux; 4.9.0-6-amd64; c8e49420-13ae-11e9-ad90-000c29e7cc9f)
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] With post data: {"params": ["magnet:?xt=urn:btih:84C925462E92AB73EDB07AD2F8C7D2D10C812B12&dn=Dinotrux.Supercharged.S02E03.720p.W$
2019-01-11 07:58:57 DEBUG    SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Response to POST request is {"id": 2, "result": null, "error": null}
2019-01-11 07:58:57 WARNING  SNATCHQUEUE-SNATCH-337661 :: [918cfe7] Deluge: Unable to send Torrent
```
</details>

**Additional context**

**Comment by fuxxociety** [comment]
I agree. But it is, somehow.

I made a copy of my main.db to see when the release was first snatched. I copied the infohash for the last error in my posted logfile snippet. I don't know if the sqlite date string is YYYY/MM/DD or YYYY/DD/MM, but either way, it was initially snatched back in 2018.

Attached the resulting line from sqlite.
![db_snippet](https://user-images.githubusercontent.com/3860592/51066366-503b7c00-15cf-11e9-8806-88dcb2f66d3f.png)
